programs/drift/src/controller/pnl.rs

```diff
@@ -237,25 +257,31 @@ pub fn settle_pnl(
         user_unsettled_pnl,
         now,
     )?;
+
     if user_unsettled_pnl == 0 {
-        msg!("User has no unsettled pnl for market {}", market_index);
-        return Ok(());
+        let msg = format!("User has no unsettled pnl for market {}", market_index);
+        return mode.result(ErrorCode::NoUnsettledPnl, &msg);
```

Comment:
is it bad these dont always return OK? do we need `update_pool_balances` to go through above?

Reply:
only the 'MUST' will cause `update_pool_balances` to not go through? think its fine to just have 'TRY' allow that